### PR TITLE
API testing

### LINKED ISSUE #34: Unit/integration testing
I am aware that we _do_ have a `test` directory, but I think it would be in our best interest to start using [Mocha](http://mochajs.org/) or another framework for unit testing. Hopefully, this would stop us pushing broken stuff into production because we don't have the testing infrastructure.

### PR DESCRIPTION
**Pull request by nickbclifford**
I am _so_ excited for this.
This PR sets up testing infrastructure with Mocha and Chai and adds tests for almost every production-facing route (some had to be skipped for practicality), in addition to Travis integration.
Closes #34.

### CODE REVIEW

File: src/libs/errors.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export class InputError extends Error {
```
Comment (MeMeBigProgrammer) [inline src/libs/errors.ts]: Is this error for Input from the User or Input in general?

File: src/libs/schedule.ts [hunk 2]
```diff
@@ -282,51 +282,53 @@ async function getSchedule(db: Db, user: string, date: Date, portalBroke = false
...
+			const start = moment(calEvent.start);
+			const end = moment(calEvent.end);
 
-		// Check if event occurs on specified day
-		if (scheduleDate.isSame(start, 'day')) {
-			// Check if special schedule
-			const lowercaseSummary = calEvent.summary!.toLowerCase();
-			if (lowercaseSummary.includes('special') && lowercaseSummary.includes('schedule')) {
-				schedule.special = true;
+			// Make sure the event isn't all whacky
```
[the remark this answers is not in the document]
Comment (nickbclifford) [reply src/libs/schedule.ts]: It wouldn't make sense for a calendar event to have an end date before a start date, so it's just checking for that (that was in the pre-PR code, git just shows it in the diff cause I messed with the indentation)

File: src/libs/errors.ts (hunk 1 above)
Comment (nickbclifford) [reply src/libs/errors.ts]: It's for any input that is received as part of a request, basically anything that we should raise a 400 error on.

Comment (MeMeBigProgrammer) [reply src/libs/errors.ts]: for both of these, adding a comment to specify that would be helpful in my opinion.

**Pull request merged**